tests/Aspire.Hosting.Docker.Tests/DockerComposePublisherTests.cs

```diff
@@ -33,8 +33,17 @@ public async Task PublishAsync_GeneratesValidDockerComposeFile()
         // Add a container to the application
         var redis = builder.AddContainer("cache", "redis")
                     .WithEntrypoint("/bin/sh")
+                    .WithHttpEndpoint(name: "h2", port: 5000, targetPort: 5001)
+                    .WithHttpEndpoint(env: "REDIS_PORT")
                     .WithArgs("-c", "hello $MSG")
-                    .WithEnvironment("MSG", "world");
+                    .WithEnvironment("MSG", "world")
+                    .WithEnvironment(context =>
+                    {
+                        var resource = (IResourceWithEndpoints)context.Resource;
+
+                        context.EnvironmentVariables["TP"] = resource.GetEndpoint("http").Property(EndpointProperty.TargetPort);
```

Comment:
`resource.GetEndpoint("http")` is the endpoint registered by the call to `.WithHttpEndpoint(env: "REDIS_PORT")` correct?

Reply:
Yep